### DISCUSSION TITLE
HTTPS support for Influxdb

### DISCUSSION THREAD

**Discussion by Derek-K** [arendst/Tasmota#19067] [Support request]
I was excited to see InfluxDB support for Tasmota, however, after much research (e.g. enabling USE_WEBCLIENT_HTTPS), it turns out HTTPS is not supported as per @arendst comment

https://github.com/arendst/Tasmota/issues/16204#issuecomment-1211919782

Can HTTPS support be added with an optional flag to include in the custom build? Thanks.

**Comment by barbudor** [comment]
Https won't be enabled in ESP8266 for memory constraints
However on ESP32 it should work.
Which are you using?

**Reply by sfromis** [reply]
IIRC, the InfluxDB driver on ESP32 only checks `USE_WEBCLIENT_HTTPS` to use an enhanced http library, [while the URL string is still only built for the http protocol](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_59_influxdb.ino#L126).

**Comment by Derek-K** [comment]
> Https won't be enabled in ESP8266 for memory constraints However on ESP32 it should work. Which are you using?

I am planning to use it with ESP32.



> IIRC, the InfluxDB driver on ESP32 only checks `USE_WEBCLIENT_HTTPS` to use an enhanced http library, [while the URL string is still only built for the http protocol](https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_59_influxdb.ino#L126).

Yea, that's how I found out it doesn't work even with `USE_WEBCLIENT_HTTPS` :sweat_smile: Hoping the author (or someone) has the kung-fu to add https feature back in, perhaps with an optional flag to compile into the binary for those who needed.

**Reply by barbudor** [reply]
Finally fixed now
InfluxDB on ESP32 support https

**Reply by nikito7** [reply]
In ESP32-C6 webui have timeouts.

Do I need a dual core?

With http it's ok

**Reply by s-hadinger** [reply]
Dual core is not supported by Tasmota nor Arduino framework. It will not change much

**Reply by nikito7** [reply]
Same issue with ESP32-S3

**Reply by barbudor** [reply]
What issue ?
You have an issue with InfluxDB over HTTPS ?
This is what this topic is about

**Reply by nikito7** [reply]
> What issue ? You have an issue with InfluxDB over HTTPS ? This is what this topic is about

Webui timeouts or stop update sensors in main page

I'm talking about influxdb http vs https in last message

**Reply by barbudor** [reply]
So that's nothing to do with the topic : "HTTPS support for Influxdb"
Don't hijack threads with comments that have nothing to

**Reply by nikito7** [reply]
Why not? InfluxDB https cause webui to timeout.

Maybe it's a webui problem or a InfluxDB code problem.

InfluxDB it's working in blocking mode.

Maybe there is no another way.

Sorry for disturb you.
